### PR TITLE
Lambda functions for app matching

### PR DESCRIPTION
**Pull request by adamroach**
This is still pretty rough, and not intended to be merged as-is. It is intended to be useful as the basis for discussing whether this is the right approach to handle Issue #63. It will need further refining if we decide to pursue this approach.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1011,6 +1041,106 @@
...
+          not considered to be a candidate to handle the
+          <code>PaymentRequest</code> and terminate these steps.
+        </li>
+        <li>
+          If the <code>PaymentAppManifest</code> does not contain a
+          <code>canHandle</code> attribute, then this <a>payment app</a> is a
+          candidate for handling the <code>PaymentRequest</code>. Only those
+          <a>PaymentAppOption</a>s that have <code>enabledMethods</code>
+          containing at least one member of <var>commonMethods</var> are to be
+          displayed to the user. Terminate these steps.
```
Comment (rsolomakhin) [inline index.html:1097]: Can we rephrase "to be displayed to the user" as "to be made available to the user for selection"? Chrome displays only one PaymentAppOption at first, with others available in a dropdown.

**Review by rsolomakhin — COMMENTED** [review]
This looks good, sans our conversation about returning a sequence of dictionaries instead of an object.

**Comment by ianbjacobs** [comment]
+1 to "to be made available to the user for selection"

**Comment by adrianhopebailie** [comment]
Really like this feature! Thanks @adamroach and @rsolomakhin for putting it together

**Comment by dlongley** [comment]
+1

**Pull request merged**